internal/gke/create.go

```diff
@@ -272,7 +273,7 @@ func validateNodePoolCreateRequest(clusterName string, np *gkev1.GKENodePoolConf
 	if np.InitialNodeCount == nil {
 		return fmt.Errorf(nodePoolErr, "initialNodeCount", *np.Name, clusterName)
 	}
-	if np.MaxPodsConstraint == nil {
+	if np.MaxPodsConstraint == nil && config.Spec.IPAllocationPolicy.UseIPAliases {
```

Comment:
I think we should check that `config.Spec.IPAllocationPolicy != nil` here because it may not be set for imported clusters.

Reply:
done